test/e2e/upgrade/upgrade_validation_test.go

```diff
@@ -0,0 +1,182 @@
+		crdList := &apiextensionsv1.CustomResourceDefinitionList{}
+		gomega.Expect(k8sClient.List(ctx, crdList)).To(gomega.Succeed())
+
+		for _, crd := range crdList.Items {
+			if !strings.HasPrefix(crd.Name, "kueue.x-k8s.io") {
+				continue
+			}
+			if crd.Spec.Conversion == nil || crd.Spec.Conversion.Strategy != apiextensionsv1.WebhookConverter {
+				continue
+			}
+			if crd.Spec.Conversion.Webhook == nil || crd.Spec.Conversion.Webhook.ClientConfig == nil {
+				continue
+			}
+
+			caBundle := crd.Spec.Conversion.Webhook.ClientConfig.CABundle
+			gomega.Expect(caBundle).NotTo(gomega.BeEmpty(), "CA bundle should not be empty for CRD %s", crd.Name)
+			gomega.Expect(string(caBundle)).To(gomega.ContainSubstring("-----BEGIN CERTIFICATE-----"),
+				"CA bundle should contain a valid PEM certificate for CRD %s", crd.Name)
+		}
+	})
+
+	ginkgo.It("should read pre-existing LocalQueue without x509 errors", func() {
+		lq := &kueue.LocalQueue{}
+		lqKey := types.NamespacedName{Name: "upgrade-test-lq", Namespace: "default"}
+		gomega.Expect(k8sClient.Get(ctx, lqKey, lq)).To(gomega.Succeed())
+	})
+
+	ginkgo.It("should read pre-existing ClusterQueue without x509 errors", func() {
+		cq := &kueue.ClusterQueue{}
+		cqKey := types.NamespacedName{Name: "upgrade-test-cq"}
+		gomega.Expect(k8sClient.Get(ctx, cqKey, cq)).To(gomega.Succeed())
+	})
+
+	ginkgo.It("should have webhook secret with valid certificates", func() {
+		secret := &corev1.Secret{}
+		secretKey := types.NamespacedName{Name: "kueue-webhook-server-cert", Namespace: "kueue-system"}
+		gomega.Expect(k8sClient.Get(ctx, secretKey, secret)).To(gomega.Succeed())
+
+		gomega.Expect(secret.Data).To(gomega.HaveKey("tls.crt"))
+		gomega.Expect(secret.Data).To(gomega.HaveKey("tls.key"))
+		gomega.Expect(secret.Data["tls.crt"]).NotTo(gomega.BeEmpty())
+		gomega.Expect(secret.Data["tls.key"]).NotTo(gomega.BeEmpty())
+	})
+
+	ginkgo.It("should have healthy controller manager pods", func() {
+		podList := &corev1.PodList{}
+		listOpts := []client.ListOption{
+			client.InNamespace("kueue-system"),
+			client.MatchingLabels{"control-plane": "controller-manager"},
+		}
+		gomega.Expect(k8sClient.List(ctx, podList, listOpts...)).To(gomega.Succeed())
+		gomega.Expect(podList.Items).NotTo(gomega.BeEmpty())
+
+		for _, pod := range podList.Items {
+			gomega.Expect(pod.Status.ContainerStatuses).NotTo(gomega.BeEmpty())
+			for _, containerStatus := range pod.Status.ContainerStatuses {
+				if containerStatus.Name == "manager" {
+					gomega.Expect(containerStatus.Ready).To(gomega.BeTrue(),
+						"Container %s in pod %s should be ready", containerStatus.Name, pod.Name)
+					ginkgo.GinkgoLogr.Info("Pod health check",
+						"pod", pod.Name,
+						"container", containerStatus.Name,
+						"restarts", containerStatus.RestartCount,
+						"ready", containerStatus.Ready)
+				}
+			}
+		}
+	})
+
+	ginkgo.It("should admit jobs using pre-existing LocalQueue", func() {
+		const jobNamespace = "default"
```

Comment:
We should generate a namespace for this test and tear it down.

Reply:
what are the downsides of using default here @kannon92?

I think we normally create namespaces for debuggability, but in this case the testing Job is  very limited in scope anyway.

In any case I think it is something we can cleanup in a follow up. I think the priority now is to have the test which proves it is fixed, have release candidate for testing by early users

Reply:
At least for normal distributions default is usually locked down. I was thinking that for openshift we may have difficulities submitting jobs to default namespace unless the client has admin access.

But I'm fine to iterate on this if we need to fix this.